src/overlays/actors/ovl_Bg_Mizu_Water/z_bg_mizu_water.c

```diff
@@ -21,19 +43,307 @@ const ActorInit Bg_Mizu_Water_InitVars = {
     (ActorFunc)BgMizuWater_Update,
     (ActorFunc)BgMizuWater_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mizu_Water/func_8089F8B0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mizu_Water/func_8089F9D0.s")
+static f32 unused1 = 0;
+static f32 unused2 = 110.0f;
+
+static u32 waterBoxIDs[] = { 2, 3, 5, 7, 12, 20, 21, 22 };
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F(scale, 1, ICHAIN_STOP),
+};
+
+u32 BgMizuWater_GetWaterLevelActionIndex(s16 switchFlag, GlobalContext* globalCtx) {
+    u32 ret;
+
+    if (bREG(0) != 0) {
+        switch (bREG(1)) {
+            case 0:
+                Flags_SetSwitch(globalCtx, 0x1C);
+                break;
+            case 1:
+                Flags_SetSwitch(globalCtx, 0x1D);
+                break;
+            case 2:
+                Flags_SetSwitch(globalCtx, 0x1E);
+                break;
+        }
+        bREG(0) = 0;
+    }
+    if ((Flags_GetSwitch(globalCtx, 0x1C)) && (switchFlag != 0x1C)) {
+        ret = 3;
+    } else if ((Flags_GetSwitch(globalCtx, 0x1D)) && (switchFlag != 0x1D)) {
+        ret = 2;
+    } else if ((Flags_GetSwitch(globalCtx, 0x1E)) && (switchFlag != 0x1E)) {
+        ret = 1;
+    } else {
+        ret = 0;
+    }
+
+    return ret;
+}
+
+void BgMizuWater_SetWaterBoxesHeight(WaterBox* waterBoxes, s16 height) {
+    u32 i;
+
+    for (i = 0; i < 8; i++) {
+        waterBoxes[waterBoxIDs[i]].unk_02 = height;
+    }
+}
+
+void BgMizuWater_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgMizuWater* this = THIS;
+    f32 initialActorY;
+    WaterBox* waterBoxes;
+    s32 waterLevelActionIndex;
+
+    waterBoxes = globalCtx->colCtx.stat.colHeader->waterBoxes;
+    this->type = this->actor.params & 0xFF;
+    this->switchFlag = (this->actor.params >> 8) & 0xFF;
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    initialActorY = this->actor.posRot.pos.y;
+    this->baseY = initialActorY;
+    this->targetY = initialActorY;
```

Comment:
Any chance you could get rid of `initialActorY` like this? (it will probably have to be replaced by a pad)
```suggestion
    this->targetY = this->baseY =  this->actor.posRot.pos.y;
```

Reply:
Sadly no, `initialActorY` is required.